project/Build.scala

```diff
@@ -504,15 +504,17 @@ object Build {
         val args: Seq[String] = spaceDelimited("<arg>").parsed
         val java: String = Process("which" :: "java" :: Nil).!!
         val attList = (dependencyClasspath in Runtime).value
-        val _  = packageAll.value
+        val _ = packageAll.value
         val scalaLib = attList
           .map(_.data.getAbsolutePath)
           .find(_.contains("scala-library"))
           .toList.mkString(":")
 
-        if (java == "")
+        if (args.isEmpty) {
+          println("Couldn't run `dotr` without args. Use `repl` to run the repl or add args to run the dotty application")
```

Comment:
Why not call the repl task at this point instead?

Reply:
I tried with `repl.evaluated` but then `dotr` only ran `repl` ignoring the nonempty args. Then I just decided to show this message as we don't really need more in the sbt project.

Reply:
sbt task evaluation is tricky, you probably need to use a dynamic task to get that to work.

Reply:
That is what I thought, hence settled for the message. My main concerns were the factorisation of code and checking the it was executing correctly.

Reply:
I don't think we should special case for empty `args`. We need a better solution:
If I run `dotr -Xprint:typer`, `args` is not empty, but I still want to run the REPL

Reply:
you can at least run `repl -Xprint:typer`.